File: TypeScript/codewars/Typescript/Disarium-Number-Special-Numbers-Series-3.ts
export function disariumNumber(n: number): string {
	const result = String(n)
		.split("")
		.map(Number)
		.reduce((sum, num, index) => sum + num ** (index + 1), 0);

	return n === result ? "Disarium !!" : "Not !!";
}

console.log(disariumNumber(564));

File: Csharp/CodeWars/src/Resolutions/DisariumNumberSpecialNumbersSeries3.cs
namespace CodeWars.Resolutions;

public class DisariumNumberSpecialNumbersSeries3
{
    public static string DisariumNumber(int number) =>
        number
        .ToString()
        .Select((value, index) => Math.Pow(char.GetNumericValue(value), index + 1))
        .Sum() == number
        ? "Disarium !!"
        : "Not !!";
}
